fluss-server/src/main/java/org/apache/fluss/server/RpcServiceBase.java

```diff
@@ -143,6 +155,16 @@ public abstract class RpcServiceBase extends RpcGatewayService implements AdminR
     private long tokenLastUpdateTimeMs = 0;
     private ObtainedSecurityToken securityToken = null;
 
+    private static final ExecutorService ZK_META_UPDATE_EXECUTOR =
+            Executors.newFixedThreadPool(8, new ExecutorThreadFactory("zk-metadata-update-"));
+
+    private static final ConcurrentHashMap<
+                    TableMetadataKey, CompletableFuture<List<BucketMetadata>>>
+            PENDING_TABLE_METADATA_FROM_ZK_FUTURES = MapUtils.newConcurrentHashMap();
+
+    private static final ConcurrentHashMap<PhysicalTablePath, CompletableFuture<PartitionMetadata>>
+            PENDING_PARTITION_METADATA_FROM_ZK_FUTURES = MapUtils.newConcurrentHashMap();
+
```

Comment:
We should avoid introducing global resources which is hard to maintain (resource leak in production) and mock (override in tests). In order to support async zk operations, we can learn from Kafka that:

1. Use the default executor service in zk client (`curatorFramework.inBackground(...)` in our case)
2. Extend the `ZookeeperClient` to support **async batch** operations. We have already introduced a basic implementation in https://github.com/apache/fluss/commit/8b9d2937c35f47dce0d034785ee684d49ba04bc3, which is easy to extend for more methods. 
3. Use the **async batch** API of `ZookeeperClient` for fetching metadatas of table/partition lists.